airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -359,3 +369,15 @@ def ti_put_rtif(
     _update_rtif(task_instance, put_rtif_payload, session)
 
     return {"message": "Rendered task instance fields successfully set"}
+
+
+def _is_eligible_to_retry(state: str, try_number: int, max_tries: int) -> bool:
+    """Is task instance is eligible for retry."""
+    if state == State.RESTARTING:
+        # If a task is cleared when running, it goes into RESTARTING state and is always
+        # eligible for retry
+        return True
```

Comment:
We should add a test for the `RESTARTING` too

Reply:
Added in b230ff61144ec3ebcf64edf005b4ea9ef931bef4